pkg/storage/db.go

```diff
@@ -151,10 +152,35 @@ func (db *DB) Init(ctx context.Context, postgresConfig *PostgresConfig) error {
 
 		}
 		initerr = db.RunScript(ctx, createScript)
+
+		if initerr == nil {
+			db.startPartitionGeneratorTimer(ctx)
+		}
 	})
 	return initerr
 }
 
+func (db *DB) startPartitionGeneratorTimer(ctx context.Context) {
```

Comment:
Since the lifecycle of this process is different than that of the parent context, you should create a new context here and copy the values from the parent context. I think @kconwayatlassian wrote some [code](https://github.com/asecurityteam/serverfull/blob/master/pkg/handlers/v1/invoke.go#L25) recently to make the copy process pretty easy.

Also, since this is not spawning from a request, I don't think a logger or stater will have been injected. We will likely need to inject a logger/stater instance into the parent context from main.go. It will essentially be mimicking [this](https://github.com/asecurityteam/logevent/blob/b32737d8d51c86c030b126ef9028cb2bab259aa1/http/middleware.go#L24) behavior